pkg/services/control/server/object_revive.go

```diff
@@ -0,0 +1,52 @@
+package control
+
+import (
+	"context"
+
+	"github.com/nspcc-dev/neofs-node/pkg/services/control"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (s *Server) ReviveObject(_ context.Context, request *control.ReviveObjectRequest) (*control.ReviveObjectResponse, error) {
+	err := s.isValidRequest(request)
+	if err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	// check availability
+	err = s.ready()
+	if err != nil {
+		return nil, err
+	}
+
+	var addr oid.Address
+	err = addr.DecodeString(string(request.GetBody().GetObjectAddress()))
```

Comment:
Wait, but the idea was exactly to avoid `DecodeString`.

Reply:
But I don't really understand how to avoid it. I did it like [drop object](https://github.com/nspcc-dev/neofs-node/blob/77404cc7f103c82e3ebcf93e8dee61b8270e523d/pkg/services/control/server/gc.go#L36), that @carpawell provided as example:
https://github.com/nspcc-dev/neofs-node/blob/5567e5fef477bbf9085124a6fb5e44a32fa42e35/pkg/services/control/service.proto#L123-L125
And I can't find a method to convert bytes to an address.

Reply:
Looks like our control API is broken all over the place wrt this problem, so we can just follow the way it currently works and solve the problem for all requests in #2980.